tst/testinstall/MatrixObj/testmatobj_new.g

```diff
@@ -0,0 +1,422 @@
+        getSourceOfTruth := x -> NewMatrix(opt.sourceOfTruthFilter, x[4], x[3], x[1]);
+    else 
+        getSourceOfTruth := x -> x[1];
+    fi;
+ 
+    result := [];
+
+    for ex in examples do
+        Add(result, rec(matObj := NewMatrix(filter, ex[4], ex[3], ex[1]),
+            mat := ex[1],
+            sourceOfTruth := getSourceOfTruth(ex)));
+        if BaseDomain(Last(result).matObj) <> ex[4] or NumberRows(Last(result).matObj) <> ex[2] or NumberColumns(Last(result).matObj) <> ex[3] or ConstructingFilter(Last(result).matObj) <> filter then 
+            ErrorNoReturn(Concatenation("Error while generating examples. In a case with NrRows = ", String(ex[2], " NrCols = ", String(ex[3]), " domain = ", String(ex[4]), "and filter := ", String(filter), " NumberRows, NumberColumns, BaseDomain or ConstructingFilter does not work.")));
+        fi;
+    od;
+
+    return result;
+end;
+
+# This is a concrete test function. It gets an example and should test the
+# function it specifies in its identifier. Here the function tests MatElm that
+# is element access.
+MatObjTest_TestMatElm := function(ex)
+    local col, row, elm;
+
+    for row in [1..NrRows(ex.matObj)] do
+        for col in [1..NrCols(ex.matObj)] do
+            elm := ex.matObj[row, col];
+            if elm <> ex.sourceOfTruth[row,col] then
+                return false;
+            fi; 
+        od;
+    od;
+
+    return true;
+end;
+
+MatObjTest_TestSetMatElm := function(ex_in)
+    local col, row, elm, mat, ex;
+
+    ex := ShallowCopy(ex_in);
+    mat := ex.matObj;
+    elm := Zero(BaseDomain(mat));
+    ex.sourceOfTruth[row, col] := elm;
+    ex.mat[row, col] := elm;
+
+    for row in [1..NrRows(ex.matObj)] do
+        for col in [1..NrCols(ex.matObj)] do
+            ex.matObj[row,col] := elm;
+            ex.sourceOfTruth[row,col] := elm;
+            if ex.matObj[row,col] <> ex.sourceOfTruth[row,col] then
+                return false;
+            fi; 
+        od;
+    od;
+
+    return true;
+end;
+
+MatObjTest_TestIsIdentityMat := function(ex_in)
+    #todo
+end;
+
+MatObjTest_TestInverse := function(ex_in)
+    local ex, mat, inv;
+
+    ex := ShallowCopy(ex_in);
+    mat := ex.MatObj;
+    inv := mat^(-1);
+    if IsIdentityMat(mat*inv) and IsIdentityMat(inv*mat) then 
+        return true;
+    fi;
+
+    return false;
+end;
+
+# This function requires a filter defining a MatrixObject and a record of 
+# options. 
+# This function runs tests to check whether the specified MatrixObject
+# implementation implements required methods and if these and other methods are 
+# correct (in the sense that they give correct results for the cases tested 
+# here).
+# The following options are supported:
+# randomSource:
+#   See TestMatrixObj_GenerateExample
+# state:
+#   See TestMatrixObj_GenerateExample
+# dimensions: 
+#   See TestMatrixObj_GenerateExample
+# dimensions_red: A list of tuples defining dimensions for matrices. If not set
```

Comment:
```suggestion
# dimensions_red:
#   A list of tuples defining dimensions for matrices. If not set
```
Instead of "If not set" here and "By default this is set" elsewhere and so on, I'd suggest to settle on a uniform way to describe defaults. E.g. at the end of a description, add, in a line of its own:
```
#   Default value: XYZ
```